### ISSUE TITLE
harness_adapter: _FingerprintCapture allocates [N, S, H] per layer — OOMs at the scale it's meant to benchmark

### ISSUE THREAD

**Issue by AlliedToasters** [AlliedToasters/fpwap#20]
## Context

`scripts/harness_adapter.py::_FingerprintCapture` preallocates one `[N, S, H]` CPU tensor per captured layer:

\`\`\`python
self.acts: dict[int, Tensor] = {
    i: torch.zeros(n_samples, seq_len, hidden_size, dtype=dtype)
    for i in layers
}
\`\`\`

At dataset scale this is the exact `[N, S_max, H × L]` landmine that sank lmprobe's `HiddenStateCapture` on the same benchmark workload (the consumer team OOM-killed on 10k × 2048 × 32 layers × 8192 H × bf16 = ~85 GB in CPU RAM, before any extraction results could be saved).

Concrete numbers the harness is meant to drive fpwap to:
- 70B, `layers=list(range(80))`, N=4096, seq_len=256 → 80 × 4096 × 256 × 8192 × 2B = **1.37 TB**
- 70B, `layers=list(range(80))`, N=10000, seq_len=2048 → ~26 TB

`scripts/harness_adapter.py::main()` defaults `hf_layers = list(range(n_hidden_layers))`, so the smoke test already captures every layer. Anyone running `uv run scripts/harness_adapter.py --model <70B> --n-samples 512` finds out the hard way.

## Why this matters for adoption

The point of `run_fpwap` is to let the consumer team swap it into `benchmark_offload_backends.py` and produce the killer-plot row at 70B × large-N. That row is precisely where the capture OOMs before the sweep finishes — so the harness can't actually measure the thing fpwap exists to prove.

## Fix options

**(a) Pool inside the capture.** A fingerprint only needs `[N, D]` worth of signal. Replace the `[N, S, H]` preallocation with either last-token pooling or on-the-fly masked-mean accumulation:

\`\`\`python
# Store [N, H] per layer (pooled), or accumulate [N, H] running mean of non-pad positions.
self.acts: dict[int, Tensor] = {
    i: torch.zeros(n_samples, hidden_size, dtype=dtype) for i in layers
}
\`\`\`

This is the right move if the only consumer of `cap.acts` is `compute_fingerprint`, which is currently the case.

**(b) Delegate to `RawActivations(last_token_only=True)`.** The library already has this callback. `compute_fingerprint` can read from `result.activations(layer, \"residual_post\")` which returns `[N, H]` pre-pooled. Saves an impl and locks the bench-adapter fingerprint on the reference callback.

Option (b) has the side benefit of making the adapter a real example of the recommended public API rather than a parallel ad-hoc capture class.

## Interaction with #19

#19 fixes the return shape of `fingerprint` (`float` → `[N, L*H]` tensor). That fix should be implemented on top of pooled capture, not on top of the `[N, S, H]` allocation — otherwise the fingerprint computation still pays the full RAM cost before collapsing.

## Done when

- `_FingerprintCapture` (or its replacement) stores at most `O(N × L × H)` CPU memory, not `O(N × S × L × H)`.
- `uv run scripts/harness_adapter.py --model <70B-id> --n-samples 512 --seq-len 256` completes without host-OOM when all 80 layers are captured.
- Ideally: the adapter uses the `RawActivations` callback directly, shrinking surface area.
